### PR TITLE
Update msgServer errors handling and add E2E tests

### PR DESCRIPTION
**Pull request by niilptr**
This PR:
 - updates msgServer errors handling, changing checks on amounts from "not zero" to "not positive",
 - adds test suites for E2E testing of queries and transactions,
 - groups query and tx files,
 - adds description and usage examples to cli commands.

### CODE REVIEW

File: x/slashrefund/keeper/msg_server.go [hunk 1]
```diff
@@ -52,8 +52,8 @@ func (ms msgServer) Deposit(goCtx context.Context, msg *types.MsgDeposit) (*type
 	}
 
 	// Check if is non-zero deposit
```
Comment (0xstepit) [inline x/slashrefund/keeper/msg_server.go]: update the comment

File: x/slashrefund/keeper/msg_server.go [hunk 2]
```diff
@@ -94,6 +94,12 @@ func (ms msgServer) Withdraw(goCtx context.Context, msg *types.MsgWithdraw) (*ty
 		return nil, valErr
 	}
 
+	// Check if validatorAddress corresponds to a validator
```
Comment (0xstepit) [inline x/slashrefund/keeper/msg_server.go]: I think we should allow to withdraw using an account that now is no more associated to a validator.

File: x/slashrefund/keeper/msg_server.go [hunk 3]
```diff
@@ -148,6 +154,12 @@ func (ms msgServer) Claim(goCtx context.Context, msg *types.MsgClaim) (*types.Ms
 		return nil, err
 	}
 
+	// Check if validatorAddress corresponds to a validator
```
Comment (0xstepit) [inline x/slashrefund/keeper/msg_server.go]: Same here for a previous validator

File: x/slashrefund/keeper/withdraw.go [hunk 4]
```diff
@@ -16,15 +16,15 @@ func (k Keeper) Withdraw(
...
+
 	deposit, found := k.GetDeposit(ctx, depAddr, valAddr)
 	if !found {
 		return sdk.NewCoin(tokens.Denom, sdk.NewInt(0)), time.Time{}, types.ErrNoDepositForAddress
 	}
 
 	depPool, found := k.GetDepositPool(ctx, valAddr)
 	if !found {
-		return sdk.NewCoin(tokens.Denom, sdk.NewInt(0)), time.Time{}, types.ErrNoDepositPoolForValidator
+		panic("found deposit but not the deposit pool")
```
Comment (0xstepit) [inline x/slashrefund/keeper/withdraw.go:36]: Add a comment to inform that when a deposit pool is removed, all deposit associated to it are also removed

File: x/slashrefund/keeper/msg_server.go (hunk 2 above)
Comment (niilptr) [reply x/slashrefund/keeper/msg_server.go]: I agree. If validator has been removed, then deposited funds or generated refunds linked to this validator would be unreachable. We should update this logic. I suggest doing it with a different pull request.

File: x/slashrefund/keeper/msg_server.go (hunk 1 above)
Comment (niilptr) [reply x/slashrefund/keeper/msg_server.go]: Nice catch, i'm working on it :)

File: x/slashrefund/keeper/withdraw.go (hunk 4 above)
Comment (niilptr) [reply x/slashrefund/keeper/withdraw.go:36]: I agree, with a comment it will be clearer.

**Comment by niilptr** [comment]
Added comments and added new subtest to test for max withdraw unbonding entries.

File: x/slashrefund/client/cli/query.go [hunk 5]
```diff
@@ -23,15 +24,359 @@ func GetQueryCmd(queryRoute string) *cobra.Command {
...
+	flags.AddPaginationFlagsToCmd(cmd, cmd.Use)
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdShowUnbondingDeposit() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show-unbonding-deposit [depositor-address] [validator-address]",
+		Short: "shows a unbonding_deposit",
```
Comment (0xstepit) [inline x/slashrefund/client/cli/query.go]: ... an unb...

**Review by 0xstepit — COMMENTED** [review]
seems perfect

**Pull request merged**